tests/ui/test-utils.test.ts

```diff
@@ -0,0 +1,121 @@
+import { getSymbolAt,getSymbolLength,getSymbolCodeAt } from "../../cocos/2d/utils/text-utils";
+test('getSymbolLength', function () {
+    let txt = 'w大熊🐻👨‍👩‍👧‍👦🐼';
+    expect(getSymbolLength('')).toEqual(0);
+    expect(getSymbolLength('w')).toEqual(1);
+    expect(getSymbolLength('中')).toEqual(1);
+    expect(getSymbolLength('👨‍👩‍👧‍👦')).toEqual(1);
+    expect(getSymbolLength(txt)).toEqual(6);
+});
+
+test('getSymbolAt', function () {
+    let txt = 'w大熊🐻👨‍👩‍👧‍👦🐼';
+    expect(getSymbolAt('', 0)).toStrictEqual('');
+    expect(getSymbolAt(txt, 0)).toStrictEqual('w');
+    expect(getSymbolAt(txt, 1)).toStrictEqual('大');
+    expect(getSymbolAt(txt, 2)).toStrictEqual('熊');
+    expect(getSymbolAt(txt, 3)).toStrictEqual('🐻');
+    expect(getSymbolAt(txt, 4)).toStrictEqual('👨‍👩‍👧‍👦');
+    expect(getSymbolAt(txt, 5)).toStrictEqual('🐼');
+    
+});
+
+test('getSymbolCodeAt', function () {
+    let txt = 'w大熊🐻👨‍👩‍👧‍👦🐼';
+    expect(getSymbolCodeAt(txt, 0)).toStrictEqual('119');
+    expect(getSymbolCodeAt(txt, 1)).toStrictEqual('22823');
+    expect(getSymbolCodeAt(txt, 2)).toStrictEqual('29066');
+    //  图形 = utf-16 编码 = 10进制 = 连接字符串
+    //  🐻 = 0xDB3D 0xDC38 = 55357 56379 = 5535756379;
+    //  👨‍👩‍👧‍👦 = 0xDB3D 0xDC68 0x200D 0xDB3D 0xDC69 0x200D 0xDB3D 0xDC67 0x200D 0xDB3D 0xDC66 = 55357 56424 8205 55357 56425 8205 55357 56423 8205 55357 56422 = 5535756424820555357564258205553575642382055535756422;
+    //  🐼 = 0xDB3D 0xDC3C = 55357 56380 = 5535756380
+    expect(getSymbolCodeAt(txt, 3)).toStrictEqual('5535756379'); // 🐻
+    expect(getSymbolCodeAt(txt, 4)).toStrictEqual('5535756424820555357564258205553575642382055535756422'); // 👨‍👩‍👧‍👦
+    expect(getSymbolCodeAt(txt, 5)).toStrictEqual('5535756380'); // 🐼
+});
+
+// test('getSymbolStartIndex', function () {
```

Comment:
Why comment these test cases?

Reply:
These functions are not exported, I exported them temporarily for testing

Reply:
Got it, could you please add some comments.

Reply:
ok